xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/redirection/RedirectClassRedirectionFilter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xpn.xwiki.internal.redirection;
+
+import java.io.IOException;
+
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.resource.ResourceReference;
+import org.xwiki.resource.ResourceReferenceManager;
+import org.xwiki.resource.entity.EntityResourceReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.internal.mandatory.RedirectClassDocumentInitializer;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.redirection.RedirectionFilter;
+import com.xpn.xwiki.web.Utils;
+
+/**
+ * Perform a redirection based on the presence of a {@code XWiki.RedirectClass} XClass.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Singleton
+public class RedirectClassRedirectionFilter implements RedirectionFilter
```

Comment:
I don't think it's right to have an implementation of RedirectionFilter with the hint "default", that's just one source of redirection and it should be associated to something like "XWiki.RedirectClass" for example.

Reply:
+1

Reply:
`XWiki.RedirectClass` hint added to the component.